cv_next/app/actions/cvs/uploadCv.ts

```diff
@@ -4,20 +4,25 @@ import { uploadCV, getCvsByUserId } from "@/server/api/cvs";
 import { transformGoogleViewOnlyUrl } from "@/helpers/cvLinkRegexHelper";
 import { redirect } from "next/navigation";
 import logger from "@/server/base/logger";
-
+import SupabaseHelper from "@/server/api/supabaseHelper";
 export interface InputValues {
   link: string;
   description: string;
-  catagoryId: number[] | null
+  catagoryId: number[] | null;
 }
 
 export const checkUploadCV = async ({
   cvData,
-  userId, // TODO: Switch to getting uid from server
 }: {
   cvData: InputValues;
-  userId: string;
 }): Promise<string | null> => {
+  const supabase = SupabaseHelper.getSupabaseInstance();
+  const connectedUser = await supabase.auth.getUser();
+  if (connectedUser.error || !connectedUser.data.user) {
+    logger.error("Error getting user");
+    return "Error getting user.";
+  }
+  const userId = connectedUser.data.user.id;
```

Comment:
I'll approve for now but add a TODO to change this to the new `getUser` function in #99